samples/net/coap_client/src/coap-client.c

```diff
@@ -196,6 +196,22 @@ static void event_iface_up(struct net_mgmt_event_callback *cb,
 	}
 
 	k_delayed_work_init(&retransmit_work, retransmit_request);
+}
+
+static void event_iface_up(struct net_if *iface, u8_t method)
```

Comment:
Suggest to change the function name to something like: send_coap_request(...)
And remove "struct net_if *iface" for you are not using it.